docs(genapi): add docs (#4475)
scaleway/docs-content: pages/generative-apis/reference-content/integrating-generative-apis-with-popular-tools.mdx

## Comparison of AI tools and libraries

The following table compares AI tools and libraries supported by Scaleway's Generative APIs:

| Tool/Library | Description | Use cases | Integration effort |
| --- | --- | --- | --- |
| [OpenAI client](#openai-client-libraries) | Popular AI library for natural language processing | Text generation, language translation, text summarization | Low |
| [LangChain](#langchain-rag-and-llm-applications) | Library for building AI applications | Inference, embeddings, document indexing and retrieval | Medium |
| [Continue Dev](#continue-dev-ai-coding-assistance) | Library for AI-powered coding assistance | Code completion, code review | Low |
| [cURL/Python](#custom-http-integrations) | Direct HTTP API calls for custom integrations | Custom applications, data processing | High |

<Message type="note">
  The integration effort is subjective and may vary depending on the specific use case and requirements.
</Message>

## OpenAI client libraries

Scaleway Generative APIs follow OpenAI's API structure, making integration straightforward. To get started, you'll need to install the OpenAI library and set up your API key.

### Configuration

> To use the OpenAI client library with Scaleway's Generative APIs, you'll need to install the required dependencies:

Reviewer: ```suggestion
To use the OpenAI client library with Scaleway's Generative APIs, first install the required dependencies:
```

```bash
pip install openai
```

> Then you'll need to set the API key and base URL in your OpenAI-compatible client:

Reviewer: ```suggestion
Then set the API key and base URL in your OpenAI-compatible client:
```

```python
from openai import OpenAI
client = OpenAI(
    base_url="https://api.scaleway.ai/v1",
    api_key="<API secret key>"
)
```
<Message type="tip">
  Make sure to replace `<API secret key>` with your actual API key.
</Message>

### Using OpenAI client for text generation

To use OpenAI client for text generation, you can create a `client.chat.completions` object and call the `create` method:
```python
response = client.chat.completions.create(
    model="llama-3.1-8b-instruct",
    messages=[{"role": "user", "content": "Tell me a joke about AI"}]
)
print(response.choices[0].message.content)
```

## LangChain (RAG & LLM applications)

LangChain is a popular library for building AI applications. Scaleway's Generative APIs support LangChain for both inference and embeddings.

<Message type="tip">

>   Refer our dedicated documentation for

Reviewer 2: ```suggestion
  Refer to our dedicated documentation for
```

  - [Implementing Retrieval-Augmented Generation (RAG) with LangChain and Scaleway Generative APIs](/tutorials/how-to-implement-rag-generativeapis/)
</Message>

## Continue Dev (AI coding assistance)

Continue Dev is a library that provides AI-powered coding assistance. Scaleway's Generative APIs support Continue Dev for code completion and more.

<Message type="tip">

>   Refer our dedicated documentation for

Reviewer 2: ```suggestion
  Refer to our dedicated documentation for
```

  - [Integrating Continue Dev with Visual Studio Code](/generative-apis/reference-content/adding-ai-to-vscode-using-continue/)
  - [Integrating Continue Dev with IntelliJ IDEA](/generative-apis/reference-content/adding-ai-to-intellij-using-continue/)
</Message>

## Custom HTTP integrations

You can interact with Scaleway's Generative APIs directly using any HTTP client.

### cURL example

To use cURL with Scaleway's Generative APIs, you can use the following command:
```bash
curl https://api.scaleway.ai/v1/chat/completions \
  -H "Authorization: Bearer <API secret key>" \
  -H "Content-Type: application/json" \
  -d '{
    "model": "llama-3.1-8b-instruct",
    "messages": [{"role": "user", "content": "What is quantum computing?"}]
  }'
```
<Message type="tip">
  Make sure to replace `<API secret key>` with your actual API key.
</Message>

### Python HTTP example

> To perform HTTP requests with Scaleway's Generative APIs, you'll need to install `requests` dependency:

Reviewer: ```suggestion
To perform HTTP requests with Scaleway's Generative APIs, install the `requests` dependency:
```
